### PR TITLE
2 - added some files for better development improvements

### LINKED ISSUE #2: development Improvements
Allows contributors to develop with multiple PHP versions without installing the PHP version locally.
This ensures that the code runs on the same basis for each developer.

- compose.yaml for the simple start
- Dockerfile to install php and phpstan as well as phpunit as phar
- Makefile to start the test in docker container
- add all additional files/directories to .gitattributes with flag export-ignore

@dereuromark I have a local branch for this improvement

### PR DESCRIPTION
**Pull request by vansari**
Fixes #2

### CODE REVIEW

File: compose.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: "4"
```
Comment (dereuromark) [inline compose.yaml:1]: Why 4

File: .gitattributes [hunk 2]
```diff
@@ -7,8 +7,16 @@
 *.png binary
 
 # Remove files for archives generated using `git archive`
-phpunit.xml.dist export-ignore
-.travis.yml export-ignore
-.gitignore export-ignore
-.gitattributes export-ignore
-.editorconfig export-ignore
```
Comment (dereuromark) [inline .gitattributes:14]: We can drop leading / usually

Comment (dereuromark) [inline .gitattributes:14]: We can drop leading / usually

File: compose.yaml (hunk 1 above)
Comment (vansari) [reply compose.yaml:1]: Sry, was a mistake from me :)

File: .gitattributes [hunk 3]
```diff
@@ -7,8 +7,18 @@
...
+phpcs.xml export-ignore
+phpstan.neon export-ignore
+phpunit.xml.dist export-ignore
+compose.yaml export-ignore
+Makefile export-ignore
+
+# Remove directories for archives
+.docker export-ignore
+.github export-ignore
+tests export-ignore
```
Comment (dereuromark) [inline .gitattributes:24]: we might want to keep tests, as they usually show usage.

File: .gitattributes [hunk 4]
```diff
@@ -7,8 +7,18 @@
...
 # Remove files for archives generated using `git archive`
-phpunit.xml.dist export-ignore
-.travis.yml export-ignore
-.gitignore export-ignore
-.gitattributes export-ignore
 .editorconfig export-ignore
+.env export-ignore
+.gitattributes export-ignore
+.gitignore export-ignore
+.travis.yml export-ignore
```
Comment (dereuromark) [inline .gitattributes:14]: this is outdated though, we dont need it

File: .gitattributes (hunk 3 above)
Comment (vansari) [reply .gitattributes:24]: Hmmm I think it should be better that the README has a high quality instead to keep a unnecessary tests directory. It blows up the archive without any usage and increase the network traffic. :)

The usage should be available in README or the code should instruct the user.

What do you think?

Comment (dereuromark) [reply .gitattributes:23]: Fair enough, but best to add trailing slash / so it is clear those are directories, not files.

**Pull request merged**